docs/control-structures.rst

```diff
@@ -51,40 +54,63 @@ write::
 
 The names of output parameters can be omitted.
 The output values can also be specified using ``return`` statements.
-The ``return`` statements are also capable of returning multiple
-values, see :ref:`multi-return`.
+
+The ``return`` statements are also capable of returning multiple values using ``return (v0, v1, ..., vn)``. The number of parameters must equal the number of output parameters.
+
 Return parameters are initialized to zero; if they are not explicitly
 set, they stay to be zero.
 
 Input parameters and output parameters can be used as expressions in
 the function body.  There, they are also usable in the left-hand side
 of assignment.
 
+.. index:: return array, return string, array, string, array of strings, dynamic array, variable sized array
+
+.. _return-array:
+
+Returning Strings and Arrays
+----------------------------
+
+You can return strings and arrays from Solidity functions, see `array_receiver_and_returner.sol <https://github.com/fivedogit/solidity-baby-steps/blob/master/contracts/60_array_receiver_and_returner.sol>`_ for an example.
+
+.. index:: return struct, struct
+
+.. _return-struct:
+
+Returning Structs
+-----------------
+
+You can return a ``struct`` from a Solidity function, but only from ``internal`` function calls.
```

Comment:
This is only the case when not using ABIEncoderV2.

Reply:
In general, I'm not sure if "returning structs" and "returning arrays" deserve their own sections. Especially if the answer is just 'yes you can'.

Reply:
@chriseth In this case, there's the note too:

> You can return a ``struct`` from a Solidity function, but only from ``internal`` function calls.

Is that still valid?

Reply:
Yes, still valid, but it is a bit more complicated: You can return memory structs from any function if `ABIEncoderV2` is activated. Furthermore, library functions also support returning storage structs.

Reply:
@chriseth OK, how about this as a combined solution…